<launch>
    <arg name="fcu_conn" default="uart"/>
    <arg name="fcu_ip" default="127.0.0.1"/>
    <arg name="gcs_bridge" default="tcp"/>
    <arg name="viz" default="true"/>

    <include file="$(find mavros)/launch/px4.launch">
        <!-- UART connection -->
        <arg name="fcu_url" value="/dev/ttyAMA0:921600" if="$(eval fcu_conn is None or fcu_conn == 'uart')"/>

        <!-- USB connection -->
        <arg name="fcu_url" value="/dev/ttyACM0" if="$(eval fcu_conn == 'usb')"/>

        <!-- sitl -->
        <arg name="fcu_url" value="udp://@$(arg fcu_ip):14557" if="$(eval fcu_conn == 'udp')"/>

        <arg name="gcs_url" value="tcp-l://0.0.0.0:5760" if="$(eval gcs_bridge == 'tcp')"/>
        <arg name="gcs_url" value="udp://@192.168.11.14:14550" if="$(eval gcs_bridge == 'udp')"/> <!-- TODO: fix -->
    </include>

    <!-- Copter visualization -->
    <include file="$(find clever)/launch/copter_visualization.launch" if="$(arg viz)"/>

    <param name="mavros/local_position/frame_id" value="local_origin"/>
    <param name="mavros/local_position/tf/send" value="true"/>
    <param name="mavros/local_position/tf/frame_id" value="local_origin"/>
    <param name="mavros/local_position/tf/child_frame_id" value="fcu"/>
    <param name="mavros/global_position/tf/send" value="false"/>
</launch>
